src/Parsers/Access/ParserCreateUserQuery.cpp

```diff
@@ -166,7 +173,9 @@ namespace
                     ///generate and add salt here
                     ///random generator FIPS complaint
                     uint8_t key[32];
-                    RAND_bytes(key, sizeof(key));
+                    if (RAND_bytes(key, sizeof(key)) != 1)
+                        throw Exception(ErrorCodes::OPENSSL_ERROR, "Cannot generate salt for password. OpenSSL error code: {}", ERR_get_error());
```

Comment:
Maybe print the error string?
```suggestion
                    {
                        auto err_code = ERR_get_error();
                        char err_str[256];
                        ERR_error_string_n(err_code, err_buf, sizeof(err_buf));
                        throw Exception(ErrorCodes::OPENSSL_ERROR, "Cannot generate salt for password. OpenSSL error: {}", err_str);
                    }
```